packages/server/postgres/migrations/1709362273000_moveFailedAuthRequest.ts

```diff
@@ -0,0 +1,64 @@
+import {FirstParam} from 'parabol-client/types/generics'
+import {Client} from 'pg'
+import {r} from 'rethinkdb-ts'
+import getPgConfig from '../getPgConfig'
+import connectRethinkDB from '../../database/connectRethinkDB'
+import getPgp from '../getPgp'
+
+export async function up() {
+  await connectRethinkDB()
+  const {pgp, pg} = getPgp()
+  const batchSize = 1000
+
+  try {
+    await r.table('FailedAuthRequest').indexCreate('time').run()
+    await r.table('FailedAuthRequest').indexWait().run()
+  } catch {}
+
+  const columnSet = new pgp.helpers.ColumnSet(['email', 'ip', 'time'], {table: 'FailedAuthRequest'})
+
+  const getNextData = async (leftBoundCursor: Date | undefined) => {
+    const startAt = leftBoundCursor || r.minval
+    const nextBatch = await r
+      .table('FailedAuthRequest')
+      .between(startAt, r.maxval, {index: 'time', leftBound: 'open'})
+      .orderBy({index: 'time'})
+      .limit(batchSize)
+      .run()
+    if (nextBatch.length === 0) return null
+    if (nextBatch.length < batchSize) return nextBatch
+    const lastItem = nextBatch.pop()
+    const lastMatchingTime = nextBatch.findLastIndex((item) => item.time !== lastItem!.time)
+    if (lastMatchingTime === -1) {
+      throw new Error(
+        'batchSize is smaller than the number of items that share the same cursor. Increase batchSize'
+      )
+    }
+    return nextBatch.slice(0, lastMatchingTime)
```

Comment:
-1 this should be `nextBatch.slice(0, lastMatchingTime + 1)`, see your scheduled job PR. I also think we don't need to have this migration at all. Because it's bound to only 1 day, if an attacker gets double the amount of tries for 1 day, that shouldn't be too risky.

Reply:
@Dschoordsch, actually it should  `nextBatch.slice(0, lastMatchingTime)`! When I first read this code, I thought that +1 was correct...then I did another migration and ran into some mysteriously missing rows. When I stepped through with a debugger I caught what was going on. See:

https://github.com/ParabolInc/parabol/pull/9502#discussion_r1510340193

Reply:
Indeed this is an ephemeral table. I would be happy to nuke this migration if you come back with a +1 comment